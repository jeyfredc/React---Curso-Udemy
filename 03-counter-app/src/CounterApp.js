
import './index.css'
import PropTypes from 'prop-types';

//Functional Components
const CounterApp = ({ value }) => {

    // handleAdd
    const handleAdd = (e) => {
        console.log('+1')
    }


    return (
        <>
        <h1>CounterApp</h1>
            <p>{ value }</p>
            <button onClick={handleAdd}> +1 </button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

export default CounterApp;
