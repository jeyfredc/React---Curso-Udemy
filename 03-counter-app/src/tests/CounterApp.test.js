import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import CounterApp from '../CounterApp';

describe('Pruebas en <CounterApp />', () => {


    test('Debe de mostrar `<CounterApp />` correctamente (hacer match con un snapshot) y sus valores por defecto', () => {
        
        const wrapper = shallow( <CounterApp /> );

        expect( wrapper).toMatchSnapshot();
    })

    test('Debe de mostrar el valor por defecto de 100', () => {
    
        const counter = '100'
        const wrapper = shallow( 
        <CounterApp 
        /> );

        const textoParrafo = wrapper.find('p').text()
        // console.log(textoParrafo); 

        expect( textoParrafo ).toBe( counter );
    })
})
