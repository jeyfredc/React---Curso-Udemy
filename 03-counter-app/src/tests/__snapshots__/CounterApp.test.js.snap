// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> Debe de mostrar \`<CounterApp />\` correctamente (hacer match con un snapshot) y sus valores por defecto 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <p>
    100
  </p>
  <button
    onClick={[Function]}
  >
     +1 
  </button>
  <button
    onClick={[Function]}
  >
     Reset
  </button>
  <button
    onClick={[Function]}
  >
     -1 
  </button>
</Fragment>
`;
